### PR TITLE
QEA-1180

### PR DESCRIPTION
**Pull request by yet-another-lucas**
Rescue errors when saving files to s3.
mako ui p1 jobs are really crudded up with errors like this:
```
Seahorse::Client::NetworkingError:
  Failed to open TCP connection to sg-qe-artifacts.s3.amazonaws.com:443 (getaddrinfo: Temporary failure in name resolution)
```

Not to mention when I invalidated the credentials recently and the same jobs caught on fire.

Also some whitespace that I cleaned up.

I don't have a test for the TCP error, but there is one for bad credentials which works. The rescue is in the same place so I'm confident it will work.

I want to revisit this and figure out how to force that error to happen in a test.

### CODE REVIEW

File: lib/s3.rb [hunk 1]
```diff
@@ -20,11 +20,19 @@ def save_file(absolute_path_of_file)
...
-            # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
-            @bucket.object(destination_name).public_url
+            begin
+              @bucket.object(destination_name).upload_file(absolute_path_of_file)
+              @bucket.object(destination_name).wait_until_exists
+              _verify_upload(destination_name, absolute_path_of_file)
+              # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
+              return @bucket.object(destination_name).public_url
+            rescue Aws::S3::Errors::InvalidAccessKeyId
+              Log.error("unable to save file to s3 due to Aws::S3::Errors::InvalidAccessKeyId")
```
Comment (sethuster) [inline lib/s3.rb]: I would like these messages to have a prefix like [GRIDIUM::S3] Message...

File: lib/s3.rb [hunk 2]
```diff
@@ -20,11 +20,19 @@ def save_file(absolute_path_of_file)
...
-            _verify_upload(destination_name, absolute_path_of_file)
-            # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
-            @bucket.object(destination_name).public_url
+            begin
+              @bucket.object(destination_name).upload_file(absolute_path_of_file)
+              @bucket.object(destination_name).wait_until_exists
+              _verify_upload(destination_name, absolute_path_of_file)
+              # @bucket.object(s3_name).presigned_url(:get, expires_in: 3600) #uncomment this if public url ends up not working out OPREQ-83850
+              return @bucket.object(destination_name).public_url
+            rescue Aws::S3::Errors::InvalidAccessKeyId
```
Comment (sethuster) [inline lib/s3.rb]: do we want to attempt a few times to upload these files?  Right now it's just rescuing at the first sign of failure and not trying again.  what are your thoughts there?

Comment (yet-another-lucas) [inline lib/s3.rb]: For the errors I've encountered, I don't think retry will work. 
The TCP error is happening when we're already waiting for the bucket, which has retry logic baked into it already.
And of course for bad credentials, they stay bad.

File: lib/s3.rb (hunk 1 above)
Comment (yet-another-lucas) [inline lib/s3.rb]: Makes total sense. Will do.

**Comment by yet-another-lucas** [comment]
Totally agree on standardizing logging. Like have the logger prefix all messages with the name of the module/class sending them. I see something called progname that goes unused in the logger.formatter that we may be able to repurpose.

I'm of two minds on the versioning. On one side, I really like separating concerns with smaller more locally scoped and atomic changes. On another side, I don't like clogging the pipes with tons of new gems. Open questions I guess: what version velocity is too high and what change set size is too small to get published alone.

**Pull request merged**